### PR TITLE
Performance: 🚀 Change ParseUrl  to use LRUCache

### PR DESCRIPTION
**Pull request by cjdeleon62**
Improves the performance of ParseUrl by allowing the function to keep track of the urls using an LRU cache with a max capacity of 100.

issue : https://github.com/ampproject/amphtml/issues/13280

### CODE REVIEW

**Comment by cjdeleon62** [comment]
I signed it!

File: test/functional/test-url.js [hunk 1]
```diff
@@ -120,6 +117,16 @@ describe('parseUrl', () => {
     const a2 = parseUrl(url);
     expect(a1).to.equal(a2);
   });
+  it('caches up to 100 results', () => {
+    const url = 'https://foo.com:123/abc?123#foo';
+    const a1 = parseUrl(url);
+    // cache 99 more urls in order to reach max capacity of LRU cache: 100
+    for (let i = 0; i < 100; i++) {
+      parseUrl(`${url}-${i}`);
```
Comment (aghassemi) [inline test/functional/test-url.js]: expecting result of parseUrl to be same as a1 would be a nice addition to ensure cache part is working as well.

Comment (cjdeleon62) [reply test/functional/test-url.js]: @aghassemi Ah right, that definitely makes sense. I'll add that in

**Comment by cjdeleon62** [comment]
@aghassemi wondering if you can review the changes I made after your comments :smiley:

**Comment by aghassemi** [comment]
LGTM, Merged. Thanks for contributing @cjdeleon62!

**Comment by cjdeleon62** [comment]
@aghassemi thanks!

**Pull request merged**